lib/mix/lib/mix/tasks/compile.elixir.ex

```diff
@@ -47,6 +50,14 @@ defmodule Mix.Tasks.Compile.Elixir do
   def run(args) do
     {opts, _, _} = OptionParser.parse(args, switches: @switches)
 
+    # The :long_compilation_threshold is in seconds, but we want to pass it to
+    # the compiler in milliseconds.
+    opts =
+      case Keyword.fetch(opts, :long_compilation_threshold) do
```

Comment:
I think we should rather make it seconds all the way up to the compiler. I
don't see why anyone would want higher precision than that?

On Sunday, May 15, 2016, Andrea Leopardi <notifications@github.com> wrote:

> In lib/mix/lib/mix/tasks/compile.elixir.ex
> <https://github.com/elixir-lang/elixir/pull/4628#discussion_r63293487>:
>
> > @@ -47,6 +50,14 @@ defmodule Mix.Tasks.Compile.Elixir do
> >    def run(args) do
> >      {opts, _, _} = OptionParser.parse(args, switches: @switches)
> >
> > +    # The :long_compilation_threshold is in seconds, but we want to pass it to
> > +    # the compiler in milliseconds.
> > +    opts =
> > +      case Keyword.fetch(opts, :long_compilation_threshold) do
>
> Good point. I updated to use Keyword.get/3 everywhere, I like it more
> than || :)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4628/files/1d6259c1607fe0e668cd911fff01fe50075c2570#r63293487>
>


-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D